pkg/controller.v2/controller.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllerv2
+
+import ctrl "sigs.k8s.io/controller-runtime"
+
+func SetupControllers(mgr ctrl.Manager) (string, error) {
+	if err := NewTrainingRuntimeReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor("training-operator-traininigruntime-controller"),
+	).SetupWithManager(mgr); err != nil {
+		return "TrainingRuntime", err
+	}
+	if err := NewClusterTrainingRuntimeReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor("training-operator-clustertraininigruntime-controller"),
+	).SetupWithManager(mgr); err != nil {
+		return "ClusterTrainingRuntime", err
+	}
+	if err := NewTrainJobReconciler(
+		mgr.GetClient(),
+		mgr.GetEventRecorderFor("training-operator-trainjob-controller"),
```

Comment:
Do we need to add the `training-operator-` name in the event recorder ?

Reply:
Yes, we should add the prefix since this event recorder name is used across the entire cluster.